examples/angular/src/app/guards/can-match-page.guard.ts

```diff
@@ -0,0 +1,42 @@
+import { inject } from '@angular/core';
+import { CanMatchFn, Route, Router, UrlSegment } from '@angular/router';
+
+import { DOTCMS_CLIENT_TOKEN } from '../client-token/dotcms-client';
+import { DotCMSPageAsset } from '@dotcms/angular';
+
+export const canMatchPage: CanMatchFn = async (
+  route: Route,
+  segments: UrlSegment[]
+) => {
+  const router = inject(Router);
+  const client = inject(DOTCMS_CLIENT_TOKEN);
+
+  const path = segments?.map((segment) => segment.path)?.join('/') || '/';
+  const initialUrl = router.getCurrentNavigation()?.initialUrl;
+  const queryParams = initialUrl?.queryParams || {};
+
+  const { mode, variantName, language_id } = queryParams;
+  const personaId = queryParams['com.dotmarketing.persona.id'] || '';
+
+  const pageProps = {
+    path,
+    mode,
+    language_id,
+    variantName,
+    'com.dotmarketing.persona.id': personaId,
+  };
+
+  try {
+    const { entity } = (await client.page.get(pageProps)) as {
+      entity: DotCMSPageAsset;
+    };
+
+    // Add the page asset to the route data
+    // so it can be used by the DotCMSPageResolver and avoid fetching it again.
+    route.data = { ...route.data, pageAsset: entity };
+
+    return !!entity;
+  } catch (error: any) {
+    return !(error?.status === 404); // If the page is not found, return false.
```

Comment:
I'm not sure, but there could be cases where this fails and is not exactly a server error.

It could fail because `entity` is not in the object of the response or a network error

what should we do in those cases?

Reply:
We can let the user handle those error cases. If we want to handle more errors in the example, we can open a new ticket. This one is for the `404` error page.

Reply:
You know what? I'll return `false` whenever it fails and log the error. That way is more robust @zJaaal! Thank you!